### ISSUE TITLE
[BUG][Regression] Marshalling error

### ISSUE THREAD

**Issue by FrancoisM** [reactiveui/ReactiveUI#2993]
**Describe the bug**

"The application called an interface that was marshalled for a different thread." for the UWP head.
Working fine with wasm. Haven't tested other heads.

**Steps To Reproduce**

Throw within a ReactiveCommand action. Subscribe to ThrownException with an action with UI impact. It throws.
Here is a [Repro](https://github.com/FrancoisM/ReproBugThreadError) based on Uno "Hello World".
Note: there are multiple commits:
- Initial commit which repro the bug with version 16.3.5
- others commits where I test with different version. 15.1.1 works, 16.1.1 fails

**Expected behaviour**

I've written this kind of code for years. It should not throw.

**Screenshots**

**Environment**

- OS:        UWP
- ReactiveUI Version:    16.3.5

**Additional context**

**Comment by glennawatson** [comment]
Might be worth trying adding "ReactiveUI.UWP" to the UWP project. We had to separate that to a different package for WinUI3 support. Let me know if there are any issues still.

**Comment by AlexisPollonni** [comment]
@glennawatson 
I don't think this issue is related with not having that package. Even with the latest version of ReactiveUI.Uwp I can reproduce the issue. I also ran into it as well while trying to update to the latest ReactiveUI versions on personal projects.
I narrowed down the code Francois provided to the minimum UWP example (Added the package you mentioned in it), and can still repro this. Should be easier to find out whats wrong:
https://github.com/AlexisPollonni/ReproBugThreadError/tree/uwp-repro-minimal-16.3.9

This is the message the exception gives when the button is pressed:
`The application called an interface that was marshalled for a different thread. (Exception from HRESULT: 0x8001010E (RPC_E_WRONG_THREAD))`

And here's the stack-trace : (not that useful i know lol)
`   at ReproThreadError.MainPageViewModel.<>c.<<-ctor>b__7_1>d.MoveNext()`

It does indeed seem like its something related with the ReactiveCommand implementation.
Hope this is a simple fix, because this prevents any UWP apps from upgrading to newer versions :)

**Comment by glennawatson** [comment]
Fixed by #2999

**Comment by Thorocaine** [comment]
Hi, I have just updated ReactiveUI from 15.1.1 to 16.3.10, and this problem is popping up for me.

I have created a [very minimal repo](https://github.com/Thorocaine/RxUiTest) to demo the problem I am having.

The line `await Task.Delay(1000).ConfigureAwait(false);` to try to force the result to return on a different thread.

Adding a breakpoint to `Label.Text = val`, When I ran this, it was on thread 14. From experience, I expect `RxApp.MainThreadScheduler` to force this onto thread 3.

Do I misunderstand what `.ObserveOn(RxApp.MainThreadScheduler)` is supposed to be doing?
This is acting as I expect it to up to version 15.1.1


```
System.Exception: 'The application called an interface that was marshalled for a different thread. (Exception from HRESULT: 0x8001010E (RPC_E_WRONG_THREAD))'
```
